boto/s3/__init__.py

```diff
@@ -59,16 +60,17 @@ def regions():
 
 
 def connect_to_region(region_name, **kw_params):
-    for region in regions():
-        if 'host' in kw_params.keys():
-            # Make sure the host specified is not nothing
-            if kw_params['host'] not in ['', None]:
-                region.endpoint = kw_params['host']
-                del kw_params['host']
-                return region.connect(**kw_params)
-            # If it is nothing then remove it from kw_params and proceed with default
-            else:
-                del kw_params['host']
-        if region.name == region_name:
+    from boto.s3.connection import S3Connection
+    if 'host' in kw_params:
+        if kw_params['host'] not in ['', None]:
+            region = S3RegionInfo(
+                name='custom',
```

Comment:
It was arbitrary. The old logic would just take whatever happened to be the first returned connection and modify it. So the region name could be 'us-east-2' or 'eu-central-1' depending on how the regions were ordered. In any case the value isn't useful here.